apps/web/src/components/chats/NftVerificationBanner.tsx

```
@@ -0,0 +1,203 @@
+'use client';
+
+import { getCurrentChainId } from '@babylon/shared';
+import { usePrivy } from '@privy-io/react-auth';
+import { Check, ExternalLink, Loader2, Shield } from 'lucide-react';
+import { useCallback, useEffect, useState } from 'react';
+import { Button } from '@/components/ui/button';
+import type { ChatDetails } from './types';
+
+interface NftVerificationBannerProps {
+  chatDetails: ChatDetails;
+  currentUserId: string | undefined;
+}
+
+interface VerificationStatus {
+  ownsNft: boolean;
+  tokenIds: number[];
+  nftRequired: boolean;
+  reason?: string;
+  contractAddress?: string;
+  tokenId?: number | null;
+  chainId?: number;
+  loading: boolean;
+}
+
+export function NftVerificationBanner({
+  chatDetails,
+  currentUserId,
+}: NftVerificationBannerProps) {
+  const { getAccessToken } = usePrivy();
+  const [verificationStatus, setVerificationStatus] =
+    useState<VerificationStatus | null>(null);
+  const [checking, setChecking] = useState(false);
+
+  const nftRequirement = chatDetails.chat.nftRequirement;
+
+  const checkVerification = useCallback(async () => {
+    setChecking(true);
+    const token = await getAccessToken();
+    if (!token) {
+      setVerificationStatus({
+        ownsNft: false,
+        tokenIds: [],
+        nftRequired: true,
+        reason: 'Authentication required',
+        loading: false,
+      });
+      setChecking(false);
+      return;
+    }
+
+    const response = await fetch(
+      `/api/chats/${chatDetails.chat.id}/nft-verification`,
+      {
+        headers: { Authorization: `Bearer ${token}` },
+      }
+    );
+
+    const data = response.ok ? await response.json() : null;
```

When `response.ok` is false, `data` is null and we default to `ownsNft: false` with no explicit error. That can read as "user does not own the NFT" rather than "verification failed". Should we surface a distinct error state/message (and keep the CTA to retry)?

---

Valid UX concern. Currently when `response.ok` is false, the component treats it as "user does not own NFT" rather than "verification failed". Adding a distinct error state with a retry CTA would improve the user experience. Marking this as a follow-up UX improvement.